Fix flake `test_read_locked_key - _queue.Empty`

### Bug summary

```
FAILED tests/records/test_filesystem.py::TestFileSystemRecordStore::test_read_locked_key - _queue.Empty
```

### Version info (`prefect version` output)

```Text
» prefect version
Version:             3.0.0rc19+9.g1a73ede315
API version:         0.8.4
Python version:      3.12.5
Git commit:          1a73ede3
Built:               Mon, Aug 26, 2024 11:39 AM
OS/Arch:             darwin/arm64
Profile:             pong
Server type:         cloud
Pydantic version:    2.8.2
```


### Additional context

_No response_

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -39,7 +39,7 @@ repos:
       - id: generate-mintlify-openapi-docs
         name: Generating OpenAPI docs for Mintlify
         language: system
-        entry: ./scripts/generate_mintlify_openapi_docs.py
+        entry: uv run --with 'pydantic>=2.9.0' ./scripts/generate_mintlify_openapi_docs.py
         pass_filenames: false
         files: |
           (?x)^(

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -21,6 +21,7 @@ pytest-xdist >= 3.6.1
 pyyaml
 redis>=5.0.1
 setuptools
+uv>=0.4.5
 vale
 vermin
 virtualenv

--- a/src/prefect/records/filesystem.py
+++ b/src/prefect/records/filesystem.py
@@ -76,7 +76,7 @@ class FileSystemRecordStore(RecordStore):
             return None
 
     def read(
-        self, key: str, holder: Optional[str] = None
+        self, key: str, holder: Optional[str] = None, timeout: Optional[float] = None
     ) -> Optional[TransactionRecord]:
         if not self.exists(key):
             return None
@@ -84,7 +84,9 @@ class FileSystemRecordStore(RecordStore):
         holder = holder or self.generate_default_holder()
 
         if self.is_locked(key) and not self.is_lock_holder(key, holder):
-            self.wait_for_lock(key)
+            unlocked = self.wait_for_lock(key, timeout=timeout)
+            if not unlocked:
+                return None
         record_data = self.records_directory.joinpath(key).read_text()
         return TransactionRecord(
             key=key, result=BaseResult.model_validate_json(record_data)

--- a/tests/records/test_filesystem.py
+++ b/tests/records/test_filesystem.py
@@ -53,23 +53,12 @@ class TestFileSystemRecordStore:
 
     async def test_read_locked_key(self, store, result):
         key = str(uuid4())
-
-        process = multiprocessing.Process(
-            target=read_locked_key,
-            args=(
-                key,
-                store,
-                (read_queue := multiprocessing.Queue()),
-            ),
-            daemon=True,
-        )
         assert store.acquire_lock(key, holder="holder1")
-        process.start()
+        assert store.read(key, holder="holder2", timeout=0) is None
         store.write(key, result=result, holder="holder1")
+        assert store.read(key, holder="holder2", timeout=1) is None
         store.release_lock(key, holder="holder1")
-        # the read should have been blocked until the lock was released
-        assert read_queue.get(timeout=10) == result
-        process.join(timeout=1)
+        assert store.read(key, holder="holder2")
 
     async def test_write_to_key_with_same_lock_holder(self, store, result):
         key = str(uuid4())

